libcontainer/init_nabla.go

```diff
@@ -23,43 +23,46 @@ import (
 	"strings"
 	"syscall"
 
+	"github.com/nabla-containers/runnc/runnc-cont"
 	spec "github.com/opencontainers/runtime-spec/specs-go"
 	"github.com/vishvananda/netns"
 )
 
 var (
-	NablaBinDir       = "/opt/runnc/bin/"
-	NablaRunncContBin = NablaBinDir + "runnc-cont"
-	NablaRunBin       = NablaBinDir + "nabla-run"
+	NablaBinDir = "/opt/runnc/bin/"
+	NablaRunBin = NablaBinDir + "nabla-run"
 )
 
-func nablaRunArgs(cfg *initConfig) ([]string, error) {
+func newRunncCont(cfg *initConfig) (*runnc_cont.RunNCCont, error) {
 	if len(cfg.Args) == 0 {
 		return nil, fmt.Errorf("OCI process args are empty")
 	}
 
 	if !strings.HasSuffix(cfg.Args[0], ".nabla") {
-		return nil, fmt.Errorf("Entrypoint is not a .nabla file")
+		return nil, fmt.Errorf("entrypoint is not a .nabla file")
 	}
 
-	args := []string{NablaRunncContBin,
-		"-k8s",
-		"-nabla-run", NablaRunBin,
-		"-tap", cfg.TapName,
-		"-cwd", cfg.Cwd,
-		"-volume", cfg.FsPath + ":/",
-		"-mem", strconv.FormatInt(cfg.Memory, 10),
-		"-unikernel", filepath.Join(cfg.Root, cfg.Args[0])}
-
-	for _, e := range cfg.Env {
-		args = append(args, "-env", e)
+	c := runnc_cont.Config{
+		NablaRunBin:    NablaRunBin,
+		UniKernelBin:   filepath.Join(cfg.Root, cfg.Args[0]),
+		Tap:            cfg.TapName,
+		IPAddress:      "10.0.0.2",
+		IPMask:         24,
+		Memory:         512,
```

Comment:
@InTheCloudDan recently added memory being passed down. Since the PR is focusing on the `runnc-cont` integration, let's not get too distracted by this. I think a small TODO comment would be perfect!